datafusion/expr/src/udwf.rs

```diff
@@ -305,10 +305,7 @@ where
 ///     .build()
 ///     .unwrap();
 /// ```
-pub trait WindowUDFImpl: Debug + Send + Sync {
-    // Note: When adding any methods (with default implementations), remember to add them also
-    // into the AliasedWindowUDFImpl below!
-
+pub trait WindowUDFImpl: Debug + DynEq + DynHash + Send + Sync {
```

Comment:
This is an API change, right? I think we should add a comment in the Upgrading guide about this

Reply:
yes; added